lib/views/commit-view.js

```diff
@@ -124,7 +124,13 @@ export default class CommitView {
   @autobind
   async commit() {
     if (await this.props.prepareToCommit() && this.isCommitButtonEnabled()) {
-      await this.props.commit(this.editor.getText());
+      await this.update({commitInProgress: true});
```

Comment:
In general, we'll want to avoid calling `update` with custom objects, since `update` is how props propagation works in Etch/React. In this case, I think it would be appropriate to add an instance variable to the *parent* component, `CommitViewController`, since that's the component that actually controls the committing (via the function it passes down to this component in the `commit` prop), and then pass its value down to this component via a property (and re-render with `etch.update(this)` in the parent as appropriate).